sites/docs/content/utilities/network-status.md

```diff
@@ -0,0 +1,104 @@
+---
+title: NetworkStatus
+description: Watch for network status changes.
+category: Browser
+---
+
+<script>
+import Demo from "$lib/components/demos/network-status.svelte";
+import { Callout } from "$lib/components";
+</script>
+
+## Demo
+
+To see it in action, try disabling and then re-enabling your Internet connection.
+If you're using Google Chrome, you can
+also [artificially throttle the network](https://developer.chrome.com/docs/devtools/settings/throttling) to test its
+behavior under different conditions.
+
+<Demo />
+
+## Usage
+
+You can use `useNetworkStatus()` to retrieve both current and previous network status.
+It must be used within the `browser` context, otherwise it will return `null`.
+
+```svelte
+<script lang="ts">
+	import { useNetworkStatus } from "runed";
+	import { toast } from "svelte-sonner";
+
+	const networkStatus = new NetworkStatus();
+	const previousNetworkStatus = new Previous(() => ({
+		online: networkStatus.online,
+		updatedAt: networkStatus.updatedAt,
+	}));
+
+	$effect(() => {
+		if (!networkStatus.isSupported) {
+			return;
+		}
+		if (networkStatus.online === false) {
+			toast.error("No internet connection.");
+		}
+		if (networkStatus.effectiveType === "2g") {
+			toast.warning("You are experiencing a slow connection.");
+		}
+		if (networkStatus.online === true && previousNetworkStatus.current?.online === false) {
+			toast.success("You are back online!");
+		}
+	});
+</script>
+
+
+{#if networkStatus.isSupported}
+    <p>online: {networkStatus.online}</p>
+{:else}
+    <p>Network Status is currently not available.</p>
+{/if}
+```
+
+### Current
+
+You can get the current status by calling the `current` method.
+
+```ts
+const networkStatus = useNetworkStatus();
+
+networkStatus.current;
+```
+
+### Previous
+
+You can get the previous status by calling the `previous` method.
+It defaults to `undefined` if the network status hasn't been updated since the component mounted.
+
+```ts
+const networkStatus = useNetworkStatus();
+
+networkStatus.previous;
+```
+
```

Comment:
todo: These sections can probably be either removed or otherwise updated to reflect the refactor

```suggestion
```

Reply:
I'll remove it and update the docs once we settle on the implementation. Currently waiting for the review from the maintainers :)